samples/README.md

```diff
@@ -18,3 +22,135 @@ for that code, and they always have the option not to.
 The hope is that samples like these may be useful to those who wish to discuss
 the spec or implement tools. To that end, we welcome comments on both the format
 of these files and their content.
+
+## Directory structure
+
+See
+[jspecify: test-data format: Directory structure](https://docs.google.com/document/d/1JVH2p61kReO8bW4AKnbkpybPYlUulVmyNrR1WRIEE_k/edit#bookmark=id.2t1r58i5a03s).
+TODO: Inline that here if Tagir can sign the CLA and contribute it.
+
+Additionally:
+
+Fully qualified class names must be unique across all directories.
+
+> This permits all files to be compiled in a single tool invocation.
+
+TODO: Consider requiring that all individual-file samples be in the top-level
+directory.
+
+TODO: Consider requiring a file's path to match its package and class:
+
+-   individual-file samples: `Foo.java` for `Foo`
+
+-   full-directory samples: `sampleFoo/Foo.java` for `Foo`,
+    `sampleFoo/bar/Foo.java` for `bar.Foo`
+
+-   We may need additional accommodations for JPMS support to demonstrate
+    module-level defaulting.
+
+## Restrictions
+
+Files must be UTF-8 encoded.
+
+Files must contain only printable ASCII characters and `\n`.
+
+Files must be compatible with Java 8.
```

Comment:
Do you mean that we can't have tests for newer language features? I suspect that local variable type inference will need some test cases

Reply:
I think, this should be rectified. Probably we should create separate packages for minimal language levels. I can imagine that some tools may be slower in adopting new language versions. In this case, then still may use part of test-suite to validate.

Reply:
Makes sense. Should I move the existing samples under a `java8` directory to make space for for future `java10`, etc. directories?

Reply:
I thought about the `java8`/`java10` split a little more, since I was also thinking about the strict/lenient split in #130. I think we may want to err on the side of using comment "tags" over directories, but I'm uncertain. I think it should be safe to leave a TODO for now, since sample inputs that use `var` don't yet exist (though we should add them sometime).

Reply:
https://github.com/jspecify/jspecify/issues/131, https://github.com/jspecify/jspecify/issues/132